x-pack/docs/en/security/securing-communications/update-tls-certificates.asciidoc

```diff
@@ -0,0 +1,574 @@
+----
+--
+
+. Stop {es} on the first node in your cluster where you are updating the
+keystore.
+
+. Replace your existing keystore with the new keystore, ensuring that the
+file names match. For example, `elastic-stack-ca.p12`.
+
+. If your keystore is password protected, the password is stored in the {es}
+keystore, _and_ the password needs to change, then update the
+`ES_PATH_CONF/elasticsearch.yml` file to use the filename of the new keystore
+file that you created.
+
+. Restart the node where you updated the keystore. 
+
+. (Optional) Use the <<security-api-ssl,SSL certificate API>> to verify that
+{es} loaded the new keystore.
+
+. On each additional node in your cluster, stop {es}, update the keystore,
+and restart {es}.
+
+[[node-certs-same-http]]
+==== Generate a new certificate for the HTTP layer
+The certificate for the HTTP layer is used to verify other components when they
+connect to {es}, such as {kib} or any of the Elastic language clients.
+
+. Stop {es} and {kib} on the first node in your cluster where you are updating
+the HTTP certificate. 
+
+. From the directory where you installed {es}, run the {es} HTTP certificate 
+tool.
++
+[source,shell]
+----
+./bin/elasticsearch-certutil http
+----
++
+This command generates a `.zip` file that contains certificates and keys
+to use with {es} and {kib}. Each folder contains a `README.txt`
+explaining how to use these files.
+
+   a. When asked if you want to generate a CSR, enter `n`.
+
+   b. When asked if you want to use an existing CA, enter `y`.
+
+   c. Enter the absolute path to your CA, such as the path to the
+    `elastic-stack-ca.p12` file.
+
+   d. Enter the password for your CA.
+
+   e. Enter an expiration value for your certificate. You can enter the
+   validity period in years, months, or days. For example, enter `90D` for 90
+   days.
+
+   f. When asked if you want to generate one certificate per node, enter `y`.
++
+Each certificate will have its own private key, and will be issued for a
+specific hostname or IP address.
+
+   g. When prompted, enter the name of the first node in your cluster. Use the
+   same node name as the value for the `node.name` parameter in the
+   `elasticsearch.yml` file.
+
+   h. Enter all hostnames used to connect to your first node. These hostnames
+   will be added as DNS names in the Subject Alternative Name (SAN) field in your certificate.
++
+List every hostname and variant used to connect to your cluster over HTTPS.
+
+   i. Enter the IP addresses that clients can use to connect to your node.
+
+   j. Repeat these steps for each additional node in your cluster.
+
+. After generating a certificate for each of your nodes, enter a password for
+   your private key when prompted.
+
+. Unzip the generated `elasticsearch-ssl-http.zip` file. This compressed file
+   contains one directory for both {es} and {kib}.
++
+--
+[source,txt]
+----
+/elasticsearch
```

Comment:
This can't be the output when we're generating all the node http certificates at once right? Don't we get one keystore for every node ?

Reply:
Correct. I think that we should use an `instances.yml` file here that mirrors the steps for generating certs with a new CA, which you mentioned in another comment.